system/lib/wasmfs/paths.cpp

```diff
@@ -11,38 +11,70 @@
 
 namespace wasmfs::path {
 
-static ParsedFile getChild(std::shared_ptr<Directory> dir,
-                           std::string_view name) {
+namespace {
+
+static inline constexpr size_t MAX_RECURSIONS = 40;
+
+ParsedFile doParseFile(std::string_view path,
+                       std::shared_ptr<Directory> base,
+                       LinkBehavior links,
+                       size_t& recursions,
+                       bool topLevel = false);
+
+ParsedFile getBase(__wasi_fd_t basefd) {
```

Comment:
The name slightly confuses me because the notation of `dirname, basename` suggests "base" to mean the child and not the dir.

How about `getBaseDir`?